### PR TITLE
docs: added deprecated option to cue

### PR DESCRIPTION
**Pull request by StephenWakely**
Quick demo of the deprecated field so we can bikeshed on what it should look like.

image

[In action](https://deploy-preview-16105--vector-project.netlify.app/docs/reference/configuration/sources/splunk_hec/#token)

This now also pulls the deprecated field through to the website. The option to add a deprecated message is also available:

```rust
    #[configurable(
        deprecated = "This option has been deprecated, the *valid_tokens* option should be used."
    )]
    token: Option<SensitiveString>,
```

Signed-off-by: Stephen Wakely <fungus.humungus@gmail.com>

### CODE REVIEW

**Comment by spencergilbert** [comment]
Red seems a _little_ "harsh", I'd put deprecations on the same level as warnings - color wise... but did we only have warnings before? I thought there was a info one as well, but I'm not seeing it in the file you changed

**Comment by neuronull** [comment]
> Red seems a _little_ "harsh", I'd put deprecations on the same level as warnings - color wise... but did we only have warnings before? I thought there was a info one as well, but I'm not seeing it in the file you changed

Was going to ask about the colors also.
 
Do we have anything else that is already red? That would help me understand how harsh red is (aside from it intrinsically being harsher than orange)

**Comment by spencergilbert** [comment]
I'm actually fine with the color - it'd normally be related to error's, which don't really make sense in this part of the documentation? Ex, the VRL function docs:

image

I don't have a good eye for color/design (imo), so I'd defer to someone who feels stronger or looked at what different built-ins we may have. Or asking some more design oriented folks at Datadog/looking at how Datadog may designate deprecated fields.

**Comment by StephenWakely** [comment]
Ok, now in yellow:

image

File: src/sources/splunk_hec/mod.rs [hunk 1]
```diff
@@ -71,7 +71,9 @@ pub struct SplunkConfig {
     /// it was communicating with the Splunk HEC endpoint directly.
     ///
     /// If _not_ supplied, the `Authorization` header will be ignored and requests will not be authenticated.
-    #[configurable(deprecated)]
+    #[configurable(
+        deprecated = "This option has been deprecated, the *valid_tokens* option should be used."
```
Comment (spencergilbert) [inline src/sources/splunk_hec/mod.rs:75]:
```suggestion
        deprecated = "This option has been deprecated, the `valid_tokens` option should be used."
```

**Review by spencergilbert — APPROVED** [review]
I'm happy to see this go forward and do whatever iteration we need design wise.

We _probably_ want to update [this text](https://deploy-preview-16105--vector-project.netlify.app/docs/reference/configuration/sources/amqp/#acknowledgements) to be in the new deprecated badge:

```text
This setting is deprecated in favor of enabling acknowledgements at the [global](https://vector.dev/docs/reference/configuration/global-options/#acknowledgements) or sink level.
```

**Review by neuronull — APPROVED** [review]
LGTM

**Comment by neuronull** [comment]
Is the goal to retro actively add this macro to the components we've already completed?

**Comment by StephenWakely** [comment]
> Is the goal to retro actively add this macro to the components we've already completed?

I think I should be able to run over them all in a single PR.

**Pull request merged**